logstash-core/lib/logstash/config/loader.rb

```diff
@@ -0,0 +1,90 @@
+require "logstash/config/defaults"
+
+module LogStash; module Config; module Loader
```

Comment:
This moves code that was in the runner into this new separate class.

This is important because new agent plugins may need to invoke this functionality themselves. For that reason the code could no longer go in either the runner or the Agent. Plus it has the happy side effect of make runner address only a single concern.